src/app/features/directory/neynar.ts

```diff
@@ -136,3 +142,32 @@ export async function getBestAddressForFid(fid: string): Promise<string> {
 
   throw new Error(`No verified addresses found for FID: ${fid}`);
 }
+
+/**
+ * Checks if a user is following another user on Farcaster
+ * @param viewerFid The FID of the viewer (current user)
+ * @param targetFid The FID of the target user
+ * @returns Promise<boolean> True if the viewer is following the target, false otherwise
+ */
+export async function isFollowing(
+  viewerFid: number,
+  targetFid: number,
+): Promise<boolean> {
+  if (!process.env.NEYNAR_API_KEY) {
+    throw new Error("NEYNAR_API_KEY is not configured");
+  }
+
+  try {
+    const result = await client.fetchUserFollowing({
+      fid: viewerFid,
```

Comment:
Is target vs viewer correct here ?

Reply:
Seems like viewer should be passed in and target should be the fid? Maybe I have it backwards but would be nice to pass both to help prioritize the relationship for paging so we don’t need to fetch multiple pages

Reply:
target is the profile.  viewer is the connected wallet app user.  Does it matter which one we initially fetch by?

Reply:
updated to use 
```
  try {
    const result = await client.fetchBulkUsers({
      fids: [targetFid],
      viewerFid: viewerFid,
    });
    return result.users[0].viewer_context?.following ?? false;
    ...
```